x-pack/plugins/enterprise_search/public/applications/shared/kibana/kibana_logic.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { ApplicationStart, ChromeBreadcrumb } from 'src/core/public';
+
+export interface IKibanaValues {
+  config: { host?: string };
+  navigateToUrl: ApplicationStart['navigateToUrl'];
+  setBreadcrumbs(crumbs: ChromeBreadcrumb[]): void;
+  setDocTitle(title: string): void;
+}
+
+export const KibanaLogic = kea<MakeLogicType<IKibanaValues>>({
+  path: ['enterprise_search', 'kibana_logic'],
+  reducers: ({ props }) => ({
+    config: [props.config || {}, {}],
+    navigateToUrl: [props.navigateToUrl, {}],
+    setBreadcrumbs: [props.setBreadcrumbs, {}],
+    setDocTitle: [props.setDocTitle, {}],
+  }),
+});
```

Comment:
Similar to the previous externalUrl PR, I also felt kinda weird that this store essentially just holds some variables that never update or require state, but in the end I decided to stay with a Kea logic file here because:

1. It's familiar
2. It's way easier to mock in tests :)

Admittedly those aren't like, super strong reasons heh, so if I'm definitely open to better paradigms if we think of them or if they come up down the road.